documentation/ContextManager.md

```diff
@@ -1,10 +1,110 @@
+
+```
+{
+    "timestamp": "2014-10-10T09:59:01.0000123+02:00",
+    "http://registry.gpii.net/common/environment/visual.luminance": 762,
+    "http://registry.gpii.net/common/environment/auditory.noise": -40
+}
+```
+
+```
+ {
+    "timestamp": "2014-23-12T09:59:01.01123923+01:00",
+    "http://registry.gpii.net/common/environment/temporal.time": "23:15"
+}
+```
+
+## Testing:
+
+To test the context managers handling of environment changes, use CURL to send a post request with the desired payload, eg:
+
+```
+curl  -H "Content-Type: application/json" -X PUT -d '{"timestamp": "2014-23-12T09:59:01.01123923+01:00","http://registry.gpii.net/common/environment/visual.luminance": 122}' http://localhost:8081/environmentChanged
+```
+
+## Supported Contexts and Condition Transformations
+
+### Supported Condition Transformations:
+
+Conditions in needs and preferences sets can come from two different sources. They can either be generated by one of the match makers, or be manually created by the user via a preferences editor.
+
+The conditions supported by the system are: 
+* `http://registry.gpii.net/conditions/inRange`
+* `http://registry.gpii.net/conditions/timeInRange`
+
+These conditions along with their expected inputs are described below:
+
+#### http://registry.gpii.net/conditions/inRange
+
+This condition is declared using the `http://registry.gpii.net/conditions/inRange` type. It can be used for any condition which deals with a range in which a context value has to fall between. Besides the context, it can take two parameters; `min` and `max`. These expresses the values (inclusive) which the `inputPath` context must fall between for the condition to be true. If one of these are omitted, that part of the range will be considered open ended (ie. if no `max` is specified, any input above or equal to `min` will be considered true)
+
+The `inputPath` can be any context that is expressed as a number. Examples of contexts that can be used with the inRange condition are `http://registry.gpii.net/common/environment/auditory.noise` and `http://registry.gpii.net/common/environment/visual.luminosity`, denoting the background noise in dB and light in luminosity.
+
+**Examples:**
+
+Example 1: Any value of visual.luminance above or equal to 700 will be considered true.
+```{
+    "type": "http://registry.gpii.net/conditions/inRange",
+    "min": 700,
+    "inputPath": "http://registry\\.gpii\\.net/common/environment/visual\\.luminance"
+}```
+
+Example 2: Any value of visual.luminance between (incl.) 600 and 1000  will be considered true.
+```{
+    "type": "http://registry.gpii.net/conditions/inRange",
+    "min": 600,
+    "max": 1000,
+    "inputPath": "http://registry\\.gpii\\.net/common/environment/visual\\.luminance"
```

Comment:
This common term (assuming it has the status of a "common term") should really be named illuminance. We are interested in incident and not emitted light.